modules/reactor/include/meshgenerators/TriPinHexAssemblyGenerator.h

```diff
@@ -0,0 +1,111 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "PolygonMeshGeneratorBase.h"
+
+/**
+ * This TriPinHexAssemblyGenerator object is a base class to be inherited for polygon
+ * mesh generators.
+ */
+class TriPinHexAssemblyGenerator : public PolygonMeshGeneratorBase
+{
+public:
+  static InputParameters validParams();
+
+  TriPinHexAssemblyGenerator(const InputParameters & parameters);
+
+  std::unique_ptr<MeshBase> generate() override;
+
+  enum USUAL_NUMBERS
+  {
+    UNITY = 1
+  };
+
+protected:
```

Comment:
any of these you can make private?

Reply:
Not sure. Maybe we can have a better idea once we need to develop a derived class for this.